### DISCUSSION TITLE
ImportError: numpy.core.multiarray failed to import #518

### DISCUSSION THREAD

**Discussion by jency-phj** [UMEP-dev/UMEP#524] [Q&A] [answered]
I also encountered the same problem, but I didn't solve it. How did you solve it later?
![image](https://github.com/UMEP-dev/UMEP/assets/71055684/c52b3c83-34d8-42b0-8a9f-2ba76464bc0d)

**Comment by biglimp** [comment]
Try by updating QGIS and UMEP.

**Reply by jency-phj** [reply]
I have tried to update qgis. I have tried the qigs version from the long-term stable version 3.28.7 to the latest version 3.30, but it still doesn't work. In addition, there is no update problem in the download of umep. After doing these things, I still can't solve this problem

**Reply by biglimp** [reply]
Ok. We have had some version conflicts between QGIS python packages and UMEP. They should have been resolved but you have probably some old numpy version on your computer. Try the following:
1. Close QGIS
2. Open the **osgeo4w shell** from your start menu
3. Type the following command: `python -m pip uninstall numpy`. Press Enter.
4. Open QGIS and try the SVF-tool again.

**Reply by jency-phj** [reply]
Thank you for your reply. I have also uninstalled, installed and updated numpy before. At that time, I found that python was version 3.10, and then numpy was the latest, which didn't match python, so I lowered the version of numpy to 1.21.*, but after this operation, I reinstalled UMEP and still reported the same error. (The specific operations are as follows: 1. Open osgeo4w and run it with administrator privileges; 2. pip show numpy to view the numpy version; 3, pip uninstall numpy to uninstall nunpy；; 4. pip show numpy reconfirms that numpy has been uninstalled; 5、pip install numpy==1.21.0; 6. Closing qgis, reopening qgis, reinstalling UMEP, and running sky module, but the result is still the same error), and this problem has not been solved. I wonder, is there something wrong with my operation? Or should I not try to change the version of numpy, but try to solve other package problems such as gdal, or is my version of qgis inappropriate?
![image](https://github.com/UMEP-dev/UMEP/assets/71055684/77164afa-0c57-4773-992e-cc8374a648b6)

**Reply by biglimp** [reply]
You should have QGIS 3.30, Python 3.9 and numpy 1.24. This is the latest versions found in the OSGeo repository. Also be aware that you have at least two locations for your numpy, both in your QGIS installation and one in your your user.

**Reply by jency-phj** [reply]
Thank you for your reply. When you mentioned numpy in the two positions mentioned, the first position is numpy in qgis. My understanding is to use osgeo4w to pip the list to view the resulting numpy; But I don't quite understand the numpy in the second position. What is' In your user '? How can I use code to verify the second numpy?

**Reply by biglimp** [reply, marked as answer]
Here is the thing: In Windows you also have a User-folder (C:\Users). In there you have a folder (**C:\Users\your_username\AppData\Roaming\Python\Python39**) where python-packages also can be located, especially if you dont have admin rights on your computer. When you install UMEP we put python packages in this location using e.g. `python -m pip install numpy -U --user`. Otherwise, people with non-admin rights would not be able to install and/or update packages. If you do `pip uninstall numpy` it might just happen that you uninstalled one of the two numpy packages and end up with version conflicts.

As I mentioned, the new version of UMEP now works with the numpy installation that comes with OSGeo.

**Reply by jency-phj** [reply]
Thank you again for your patient answer. Through your answer, I have thought many times these days, and today I have solved this problem. Speaking of which, the solution is a bit coincidental. The main point is related to what you mentioned
1. Update numpy
2. Update scipy
3. use newprofile
4. Make sure that these packages are also available in user.
5. Update netCDF4
6. Pairing numa and numpy versions
